src/AppInstallerSharedLib/SQLiteWrapper.cpp

```diff
@@ -144,49 +148,78 @@ namespace AppInstaller::SQLite
                 return {};
             }
         }
+
+        void SharedConnection::Disable()
+        {
+            m_active = false;
+        }
+
+        sqlite3* SharedConnection::Get() const
+        {
+            THROW_HR_IF(APPINSTALLER_CLI_ERROR_SQLITE_CONNECTION_TERMINATED, !m_active.load());
+            return m_dbconn.get();
+        }
+
+        sqlite3** SharedConnection::GetPtr()
+        {
+            return &m_dbconn;
+        }
     }
 
     Connection::Connection(const std::string& target, OpenDisposition disposition, OpenFlags flags)
     {
+        m_dbconn = std::make_shared<details::SharedConnection>();
         m_id = GetNextConnectionId();

```

Comment:
Is there any reason to not do these in initializer list?

Reply:
No, but there isn't really a big reason to do them in the list either.